ripd/ripd.h

```diff
@@ -99,8 +99,16 @@
 
 /* RIP structure. */
 struct rip {
-	/* VRF ID. */
-	vrf_id_t vrf_id;
+	RB_ENTRY(rip) entry;
+
+	/* VRF this routing instance is associated with. */
+	char *vrf_name;
+
+	/* VRF backpointer (might be NULL if the VRF doesn't exist). */
+	struct vrf *vrf;
+
```

Comment:
regarding that parameter, some code like bgp uses vrf_name and vrf_id.
to keep consistency, I agree that we should keep vrf pointer in all future code related to vrf.

Reply:
Yes, that's what I did at first too. But keeping track of the VRF pointer instead of the VRF id is much better if we do it correctly. I didn't like having to call `vrf_lookup_by_id(rip->vrf_id)` on lots of places so I decided to do things differently.

> I agree that we should keep vrf pointer in all future code related to vrf.

+1 to that, maybe we can also convert the other VRF-aware daemons to the new style as well in the future.